python/cuml/linear_model/linear_regression.pyx

```diff
@@ -247,7 +260,7 @@ class LinearRegression(Base):
             msg = "X matrix must have at least two rows"
             raise TypeError(msg)
 
-        if self.n_cols == 1:
+        if self.n_cols == 1 and self.algo != 0:
```

Comment:
Would be great to get an enum for this algo numbering

Reply:
It is an enum, but for Pytho we store it as an int. The translation is above in line 215: ```def _get_algorithm_int(self, algorithm): return { 'svd': 0, 'eig': 1 }[algorithm]``` Can look into changing it in a different PR

Reply:
Happy to make it a different PR. Would be nice to use a string or enum sometime though.